Client/Frontend/AuthenticationManager/PasscodeEntryViewController.swift

```diff
@@ -16,6 +16,8 @@ import SwiftKeychainWrapper
 /// Presented to the to user when asking for their passcode to validate entry into a part of the app.
 class PasscodeEntryViewController: BasePasscodeViewController {
     weak var delegate: PasscodeEntryDelegate?
+    var success: (() -> Void)?
+    var cancel: (() -> Void)?
```

Comment:
This way we can pass state more easily around, in the closure itself, rather than either assigning temporary data to the delegate, or creating a new class to deal with each unique occasion we want to use authorisation. For classes that just use the passcode for a single action, it's not a problem, but I found it to be cumbersome to use delegates when there were multiple authorisation actions originating from the same object.